src/features/common/api/api.ts

```diff
@@ -0,0 +1,15 @@
+import axios from 'axios';
+import camelcaseKeys from "camelcase-keys";
+
+
+const instance = axios.create({
+    baseURL: 'https://reqres.in/api'
+})
+
+
+//I will refactor(improve) link in this function, when add the pagination logic to project!
+export const getAllUsers = async () => {
+    const response = await instance.get<ResponseDataType>('/users');
```

Comment:
Not sure we correctly typed the response here. From what I know the `await instance.get` responds with an array so probably we should have `instance.get<ResponseDataType[]>` here?

Reply:
Also there is a problem with the current approach. We are receiving the user objects with snake_case properties e.g. { first_name: 'Andrii' }. And in the type we provided we expect the transformed objects e.g. { firstName: 'Andrii' }. It seems like we should create an additional interface|type that describes the actual response.

```js
interface UserResponse {
  first_name: string
}
```

along with it we should have a User interface|type with mapped keys

```js
interface User {
  firstName: string
}
```

This way we can rewrite the function as:

```js
async () => {
        const response = await instance.get<UserResponse>('/users');
        return camelcaseKeys({...response.data}, { deep: true }) as User ;
    }
```

Reply:
@Acousticdesk In ResponseDataType we have Array<UserType>. So we receive from server ResponseDataType and inside this type, we have an array of UserType. So I think my version is right.

Reply:
Got it, seems fine 👍 However we still should work on the correct type for User in the response